api/nodes/serializers.py

```diff
@@ -699,6 +713,20 @@ def create(self, validated_data):
             node.subjects.add(parent.subjects.all())
             node.save()
 
+        if license_details:
+            try:
+                node.set_node_license(
+                    {
+                        'id': license_details.get('id') if license_details.get('id') else 'NONE',
+                        'year': license_details.get('year'),
+                        'copyrightHolders': license_details.get('copyrightHolders') or license_details.get('copyright_holders', []),
+                    },
+                    auth=get_user_auth(request),
+                    save=True,
+                )
+            except ValidationError as e:
+                raise InvalidModelValueError(detail=e.message)
```

Comment:
Drive by: this should be stringified for Py3 raise InvalidModelValueError(detail=str(e)) or something like that.

Reply:
I'd also expect to see this this chunck of code (716-729) up around ln 660, but that's just personal preference.

Reply:
I would have too, but I was unable to figure out how to add the node license before the node itself was created

> I'd also expect to see this this chunck of code (716-729) up around ln 660, but that's just personal preference.

I would have preferred that as well, but I could not figure out how to add the node license before the node itself was created. That's why its popped off, then set later after the node is created.